src/components/FlightCard/FlightCard.tsx

```diff
@@ -1,23 +1,51 @@
+import React from "react";
 import { FlightResultsProps,  } from "../../pages/HomePage/interfaces";
-import { Button, Grid } from "@mui/material";
+import {Button, Grid, Typography} from "@mui/material";
 import flightCardStyles from "./FlightCard.module.css"
 import FlightInfoComponent from "./FlightInfoComponent";
+import FlightDetailsModal from "../FlightDetails/FlightDetailsModal";
 
-const FlightCard: React.FC<{flightResults: FlightResultsProps}> = ({flightResults} : {flightResults: FlightResultsProps}) => {
+type FlightCardProps = {
+    flightResults: FlightResultsProps,
+    showMode?: boolean
+};
+
+/**
+ *
+ * @param flightResults
+ * @param showMode - if true, the component doesn't show the price nor the details modal, if false it shows the
+ * passengers number data.
+ * @constructor
+ */
+const FlightCard: React.FC<FlightCardProps> = ({flightResults, showMode} : FlightCardProps) => {
+    const [open, setOpen] = React.useState(false);
+    const handleOpen = () => setOpen(true);
     let randomString = (Math.random() + 1).toString(36).substring(7);
```

Comment:
I would prefer to use a uuid creator or library that creates a unique uuid.

For example this one: https://www.npmjs.com/package/uuid

Reply:
This came on within back commits.
My personal opinion is that maybe we can avoid to install a whole package to not increase project bundle size and improve the random string function or simply verify if it has minor probabilities of collision